src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamCredentialCacheTest.cs

```diff
@@ -47,8 +49,18 @@ public async Task SslStream_SameCertUsedForClientAndServer_Ok()
                     // by the server using certificates from the Trusted Root Authorities certificate store.
                     // The client side will use the Trusted Issuers List, if not empty, to filter proposed certificates.
 
-                    Assert.True(client.IsMutuallyAuthenticated);
-                    Assert.True(server.IsMutuallyAuthenticated);
+                    if (clientCertificateRequired)
+                    {
+                        Assert.True(client.IsMutuallyAuthenticated);
+                        Assert.True(server.IsMutuallyAuthenticated);
+                    }
+                    else
+                    {
+                        // Even though the certificate was provided, it was not requested by the server and thus the client
```

Comment:
Will this fail on platforms where `StartMutualAuthAsAnonymous = false`?

Reply:
Most likely yes, I expected to find out during test runs on CI